freqtrade/commands/hyperopt_commands.py

```diff
@@ -35,9 +35,16 @@ def start_hyperopt_list(args: Dict[str, Any]) -> None:
         'filter_min_avg_profit': config.get('hyperopt_list_min_avg_profit', None),
         'filter_max_avg_profit': config.get('hyperopt_list_max_avg_profit', None),
         'filter_min_total_profit': config.get('hyperopt_list_min_total_profit', None),
-        'filter_max_total_profit': config.get('hyperopt_list_max_total_profit', None)
+        'filter_max_total_profit': config.get('hyperopt_list_max_total_profit', None),
+        'filter_min_objective': config.get('hyperopt_list_min_objective', None),
+        'filter_max_objective': config.get('hyperopt_list_max_objective', None)
     }
 
+    if filteroptions['filter_min_objective'] is not None:
+        filteroptions['filter_min_objective'] = -filteroptions['filter_min_objective']
+    if filteroptions['filter_max_objective'] is not None:
+        filteroptions['filter_max_objective'] = -filteroptions['filter_max_objective']
+
```

Comment:
What's the purpose of this? 
is it because objective is "negativized" in the hyperoptloss functions, too?

 (i'm asking because this caused the C901 error)

Reply:
Yes, the purpose is to multiply the given input by -1 to make it minus matching loss in hyperoptloss functions.

If you think about it, in this context, when filtering out results from the hyperopt-list, you'd consider -100 objective higher than -10 objective, so you'd use -min-objective 10 to show any epochs with -10 and lower, having it done the other way around was a bit confusing.

Reply:
> What's the purpose of this?
> is it because objective is "negativized" in the hyperoptloss functions, too?
> 
> (i'm asking because this caused the C901 error)

The C901 error was caused by the extended **_hyperopt_filter_trials()** method, not because of the code block above, to me at least.